src/components/LogInModal/LogInModal.js

```diff
@@ -0,0 +1,79 @@
+import { Link } from "react-router-dom/cjs/react-router-dom";
+import ModalWithForm from "../ModalWithForm/ModalWithForm";
+import { useState } from "react";
+import { useForm } from "../../hooks/useForm";
+import { useHistory } from "react-router-dom/cjs/react-router-dom";
+
+const LogInModal = ({
+  handleCloseModal,
+  isOpen,
+  onSubmit,
+  buttonText,
+  openRegisterModal,
+  inputError,
+}) => {
+  const { values, handleChange } = useForm({ email: "", password: "" });
```

Comment:
* According to the task you need to show validation messages under the inputs if the text is invalid

Reply:
If it’s interesting for you here is a custom `validation hook`.

You should put it in a separate file  `useFormAndValidation.js` in folder `hooks` and then import it to the component that needs validation:

```js

import {useState, useCallback} from 'react';

export function useFormAndValidation() {
  const [ values, setValues ] = useState({});
  const [ errors, setErrors ] = useState({});
  const [ isValid, setIsValid ] = useState(true);

  const handleChange = (e) => {
    const {name, value} = e.target
    setValues({...values, [name]: value });
    setErrors({...errors, [name]: e.target.validationMessage});
    setIsValid(e.target.closest('form').checkValidity());
  };



  const resetForm = useCallback((newValues = {}, newErrors = {}, newIsValid = false) => {
    setValues(newValues);
    setErrors(newErrors);
    setIsValid(newIsValid);
  }, [setValues, setErrors, setIsValid]);

  return { values, handleChange, errors, isValid, resetForm, setValues, setIsValid };
}


```

And only one line of code enables validation:

```js

  const {values, handleChange, errors, isValid, setValues, resetForm} = useFormAndValidation()

```

So, you **don’t need** to make functions for handling inputs, storing values, resetting the form. Everything is already done.